Documentation/src/case-studies/calendar.md

```diff
@@ -0,0 +1,350 @@
+        .Center(WidthOfAWeek );
+
+
+internal static class StringExtensions
+{
+    public static string Center(this string text, int width)
+        => (width - text.Length) switch
+        {
+            0 => text,
+            1 => $" {text}",
+            _ => Center($" {text} ", width),
+        };
+}
+```
+
+We have already seen the heart of FormatWeeks in the yield solution, but now it is a separate function. FormatWeeks again needs 2 very simple helper functions, the first one Projects the Week of the year, the other one will format a sequence of days.
+
+The sequence of days can be either a complete week, or a partial week from the beginning or the end of the month. But because of the way we construct these sequences, there always is at least one element in it. 
+
+```cs
+private static IEnumerable<string> FormatWeeks(IEnumerable<DateOnly> month)
+            => month
+                .AdjacentGroupBy(GetWeekOfYear)
+                .Select(FormatWeek);
+```
+
+The GetWeekOfYear function is just calling the API with the correct parameters and always using the current culture and a little fiddling with DateTime (because DateOnly was introduced far too late in C# 10).
+
+```cs
+private static int GetWeekOfYear(DateOnly dateTime)
+            => CultureInfo
+                .CurrentCulture
+                .Calendar
+                .GetWeekOfYear(dateTime.ToDateTime(default), CultureInfo.CurrentCulture.DateTimeFormat.CalendarWeekRule, CultureInfo.CurrentCulture.DateTimeFormat.FirstDayOfWeek);
+```
+
+We are almost done with `FormatMonth`, now we really format the week, each day has a width of 3 characters, we see that in FormatDay. This means each line should have the same width of 21 characters. We simply pad the week correctly with the PadWeek function.
+
+```cs
+private static string FormatWeek(IGrouping<int, DateOnly> week)
+  => PadWeek(week.Select(FormatDay).ConcatToString(), week);
+  
+private static string FormatDay(DateOnly day)
+  => $"{day.Day,WidthOfDay}";
+```
+
+We can ignore the full weeks, because they are already 21 characters long. How do we distinguish the beginning of the month from the end? The week at the end of the month must start with the first day of the week. So wee pad accordingly from left or the right.
+  
+```cs
+private static string PadWeek(string formattedWeek, IGrouping<int, DateOnly> week)
+  => StartsOnFirstDayOfWeek(week)
+    ? $"{formattedWeek,-WidthOfAWeek}"
+    : $"{formattedWeek,WidthOfAWeek}";
+```
+
+Now it just boils down to, what is the start of the week? It might be a surprise, that this is actually the most difficult part of this program, because the `DayOfWeek` enum defines Sunday as the first day of the week, which is not true for the largest part of the world, including all of europe. But `CultureInfo` has our back, because it tells us in the `DateTimeFormat`, which Day of the Week is the first Day. However even that needs some calculation with `% DaysInAWeek`.
```

Comment:
Day of the Week -> day of the week (many more capitalization errors in this paragraph)

Reply:
Fixed in 239b3a73af300f846afba75f4a992650b6fcb41b